File: lib/ui/settings/task_config_page.dart
import 'package:flutter/material.dart';
import 'package:flutter_riverpod/flutter_riverpod.dart';
import 'package:go_router/go_router.dart';
import '../../core/app_constants.dart';
import '../../providers/service_providers.dart';
import '../../providers/task_providers.dart';
import '../../services/daily_task_service.dart';

/// 基础任务设置页面
///
/// 用户可以:
/// 1. 使用推荐目标 (见人100/查询5/成交不参与)
/// 2. 自定义每个指标的目标值
/// 3. 选择是否将成交纳入基础任务
/// 4. 当天产生数据后, 目标锁定不可修改
class TaskConfigPage extends ConsumerStatefulWidget {
  const TaskConfigPage({super.key});

  @override
  ConsumerState<TaskConfigPage> createState() => _TaskConfigPageState();
}

class _TaskConfigPageState extends ConsumerState<TaskConfigPage> {
  late int _meetTarget;
  late int _queryTarget;
  late int _dealTarget;
  late bool _includeMeet;
  late bool _includeQuery;
  late bool _includeDeal;
  bool _initialized = false;
  bool _saving = false;

  @override
  Widget build(BuildContext context) {
    final configAsync = ref.watch(todayTaskConfigProvider);
    final lockedAsync = ref.watch(isTodayLockedProvider);

    if (!_initialized) {
      return configAsync.when(
        data: (config) {
          _meetTarget = config.meetTarget;
          _queryTarget = config.queryTarget;
          _dealTarget = config.dealTarget;
          _includeMeet = config.includeMeet;
          _includeQuery = config.includeQuery;
          _includeDeal = config.includeDeal;
          _initialized = true;
          return _buildScaffold(lockedAsync.valueOrNull ?? false);
        },
        loading: () => const Scaffold(
          body: Center(child: CircularProgressIndicator()),
        ),
        error: (e, _) => Scaffold(
          appBar: AppBar(title: const Text('基础任务设置')),
          body: Center(child: Text('加载失败: $e')),
        ),
      );
    }

    return _buildScaffold(lockedAsync.valueOrNull ?? false);
  }

  Widget _buildScaffold(bool locked) {
    final theme = Theme.of(context);
    return Scaffold(
        appBar: AppBar(
          title: const Text('基础任务设置'),
          leading: IconButton(
            icon: const Icon(Icons.arrow_back),
            onPressed: () => context.pop(),
          ),
        ),
        body: ListView(
          padding: const EdgeInsets.symmetric(vertical: 8),
          children: [
            // 说明
            Container(
              margin: const EdgeInsets.all(16),
              padding: const EdgeInsets.all(12),
              decoration: BoxDecoration(
                color: theme.colorScheme.primaryContainer.withValues(alpha: 0.3),
                borderRadius: BorderRadius.circular(12),
              ),
              child: Row(
                children: [
                  Icon(Icons.info_outline, color: theme.colorScheme.primary, size: 20),
                  const SizedBox(width: 8),
                  Expanded(
                    child: Text(
                      locked
                          ? '今日任务已锁定 (已产生数据), 不可修改'
                          : '设置每日最低目标, 完成全部目标即可达成今日作战',
                      style: theme.textTheme.bodySmall,
                    ),
                  ),
                ],
              ),
            ),

            // 推荐目标按钮 (仅在未锁定时显示)
            if (!locked) ...[
              Padding(
                padding: const EdgeInsets.symmetric(horizontal: 16),
                child: OutlinedButton.icon(
                  onPressed: _useRecommended,
                  icon: const Icon(Icons.recommend_outlined),
                  label: const Text('使用推荐目标 (见人100 / 查询5 / 成交不参与)'),
                ),
              ),
              const SizedBox(height: 8),
            ],

            // 每日最低目标标题
            Padding(
              padding: const EdgeInsets.fromLTRB(16, 8, 16, 4),
              child: Text(
                '每日最低目标',
                style: theme.textTheme.titleSmall?.copyWith(
                  fontWeight: FontWeight.bold,
                  color: theme.colorScheme.primary,
                ),
              ),
            ),

            // 见人数
            _MetricConfigCard(
              label: '见人数',
              icon: Icons.groups,
              color: Colors.blue,
              included: _includeMeet,
              target: _meetTarget,
              locked: locked,
              onToggleIncluded: _includeMeet
                  ? null
                  : (locked ? null : () => setState(() => _includeMeet = true)),
              onTargetChanged: locked
                  ? null
                  : (v) => setState(() => _meetTarget = v),
              onRemove: locked
                  ? null
                  : () => setState(() => _includeMeet = false),
            ),

            // 查询数
            _MetricConfigCard(
              label: '查询数',
              icon: Icons.search,
              color: Colors.purple,
              included: _includeQuery,
              target: _queryTarget,
              locked: locked,
              onToggleIncluded: _includeQuery
                  ? null
                  : (locked ? null : () => setState(() => _includeQuery = true)),
              onTargetChanged: locked
                  ? null
                  : (v) => setState(() => _queryTarget = v),
              onRemove: locked
                  ? null
                  : () => setState(() => _includeQuery = false),
            ),

            // 成交数
            _MetricConfigCard(
              label: '成交数',
              icon: Icons.celebration,
              color: Colors.red,
              included: _includeDeal,
              target: _dealTarget,
              locked: locked,
              onToggleIncluded: _includeDeal
                  ? null
                  : (locked ? null : () => setState(() => _includeDeal = true)),
              onTargetChanged: locked
                  ? null
                  : (v) => setState(() => _dealTarget = v),
              onRemove: locked
                  ? null
                  : () => setState(() => _includeDeal = false),
              isDeal: true,
            ),

            const SizedBox(height: 16),

            // 完成说明
            Container(
              margin: const EdgeInsets.symmetric(horizontal: 16),
              padding: const EdgeInsets.all(12),
              decoration: BoxDecoration(
                color: Colors.green.withValues(alpha: 0.08),
                borderRadius: BorderRadius.circular(12),
                border: Border.all(color: Colors.green.withValues(alpha: 0.2)),
              ),
              child: Row(
                children: [
                  const Icon(Icons.check_circle, color: Colors.green, size: 20),
                  const SizedBox(width: 8),
                  Expanded(
                    child: Text(
                      '完成以上全部目标即可完成今日作战\n连续完成每日基础任务 → 连续作战 +1',
                      style: theme.textTheme.bodySmall?.copyWith(
                        color: Colors.green.shade700,
                      ),
                    ),
                  ),
                ],
              ),
            ),

            const SizedBox(height: 24),

            // 保存按钮
            if (!locked)
              Padding(
                padding: const EdgeInsets.symmetric(horizontal: 16),
                child: SizedBox(
                  width: double.infinity,
                  height: 48,
                  child: FilledButton(
                    onPressed: _saving ? null : _save,
                    child: _saving
                        ? const SizedBox(
                            width: 20,
                            height: 20,
                            child: CircularProgressIndicator(
                              strokeWidth: 2,
                              color: Colors.white,
                            ),
                          )
                        : const Text('保存', style: TextStyle(fontSize: 16)),
                  ),
                ),
              ),

            const SizedBox(height: 32),
          ],
        ),
    );
  }

  void _useRecommended() {
    setState(() {
      _meetTarget = DefaultTaskConfig.recommendedMeetTarget;
      _queryTarget = DefaultTaskConfig.recommendedQueryTarget;
      _dealTarget = DefaultTaskConfig.recommendedDealTarget;
      _includeMeet = DefaultTaskConfig.recommendedIncludeMeet;
      _includeQuery = DefaultTaskConfig.recommendedIncludeQuery;
      _includeDeal = DefaultTaskConfig.recommendedIncludeDeal;
    });
  }

  Future<void> _save() async {
    if (_saving) return;

    // 验证: 至少选择一个指标
    if (!_includeMeet && !_includeQuery && !_includeDeal) {
      ScaffoldMessenger.of(context).showSnackBar(
        const SnackBar(content: Text('请至少选择一个指标')),
      );
      return;
    }

    setState(() => _saving = true);
    try {
      final taskService = ref.read(dailyTaskServiceProvider);
      final config = DailyTaskConfig(
        meetTarget: _meetTarget,
        queryTarget: _queryTarget,
        dealTarget: _dealTarget,
        includeMeet: _includeMeet,
        includeQuery: _includeQuery,
        includeDeal: _includeDeal,
      );
      await taskService.setDayConfig(DateTime.now(), config);

      // 刷新 providers
      ref.invalidate(todayTaskConfigProvider);
      ref.invalidate(isTodayLockedProvider);

      if (mounted) {
        ScaffoldMessenger.of(context).showSnackBar(
          const SnackBar(
            content: Text('已保存'),
            duration: Duration(seconds: 1),
          ),
        );
        context.pop();
      }
    } catch (e) {
      if (mounted) {
        ScaffoldMessenger.of(context).showSnackBar(
          SnackBar(content: Text('保存失败: $e')),
        );
      }
    } finally {
      if (mounted) setState(() => _saving = false);
    }
  }
}

/// 单个指标配置卡片
class _MetricConfigCard extends StatelessWidget {
  final String label;
  final IconData icon;
  final Color color;
  final bool included;
  final int target;
  final bool locked;
  final VoidCallback? onToggleIncluded;
  final ValueChanged<int>? onTargetChanged;
  final VoidCallback? onRemove;
  final bool isDeal;

  const _MetricConfigCard({
    required this.label,
    required this.icon,
    required this.color,
    required this.included,
    required this.target,
    required this.locked,
    this.onToggleIncluded,
    this.onTargetChanged,
    this.onRemove,
    this.isDeal = false,
  });

  @override
  Widget build(BuildContext context) {
    final theme = Theme.of(context);
    return Card(
      margin: const EdgeInsets.symmetric(horizontal: 16, vertical: 4),
      child: Padding(
        padding: const EdgeInsets.all(12),
        child: Column(
          children: [
            // 标题行
            Row(
              children: [
                Icon(icon, color: color, size: 22),
                const SizedBox(width: 8),
                Text(
                  label,
                  style: theme.textTheme.bodyLarge
                      ?.copyWith(fontWeight: FontWeight.w600),
                ),
                if (isDeal) ...[
                  const SizedBox(width: 6),
                  Container(
                    padding: const EdgeInsets.symmetric(horizontal: 6, vertical: 2),
                    decoration: BoxDecoration(
                      color: Colors.orange.withValues(alpha: 0.15),
                      borderRadius: BorderRadius.circular(4),
                    ),
                    child: Text(
                      '默认不参与',
                      style: theme.textTheme.labelSmall
                          ?.copyWith(color: Colors.orange.shade700),
                    ),
                  ),
                ],
                const Spacer(),
                if (included)
                  Text(
                    '参与',
                    style: theme.textTheme.labelSmall?.copyWith(
                      color: Colors.green,
                      fontWeight: FontWeight.bold,
                    ),
                  )
                else
                  Text(
                    '不参与',
                    style: theme.textTheme.labelSmall?.copyWith(
                      color: theme.colorScheme.onSurfaceVariant,
                    ),
                  ),
              ],
            ),

            if (included) ...[
              const SizedBox(height: 12),
              const Divider(height: 1),
              const SizedBox(height: 12),
              // 目标调整
              Row(
                children: [
                  Text('目标', style: theme.textTheme.bodyMedium),
                  const Spacer(),
                  // 减号
                  IconButton(
                    onPressed: locked || onTargetChanged == null
                        ? null
                        : () => onTargetChanged!(target > 1 ? target - 1 : 1),
                    icon: const Icon(Icons.remove_circle_outline),
                    color: color,
                    iconSize: 20,
                    constraints: const BoxConstraints(
                        minWidth: 36, minHeight: 36),
                    padding: EdgeInsets.zero,
                  ),
                  // 数字
                  Container(
                    width: 56,
                    padding: const EdgeInsets.symmetric(vertical: 4),
                    decoration: BoxDecoration(
                      border: Border.all(color: color.withValues(alpha: 0.3)),
                      borderRadius: BorderRadius.circular(8),
                    ),
                    child: Text(
                      '$target',
                      textAlign: TextAlign.center,
                      style: theme.textTheme.titleMedium?.copyWith(
                        fontWeight: FontWeight.bold,
                        color: color,
                      ),
                    ),
                  ),
                  // 加号
                  IconButton(
                    onPressed: locked || onTargetChanged == null
                        ? null
                        : () => onTargetChanged!(target + 1),
                    icon: const Icon(Icons.add_circle_outline),
                    color: color,
                    iconSize: 20,
                    constraints: const BoxConstraints(
                        minWidth: 36, minHeight: 36),
                    padding: EdgeInsets.zero,
                  ),
                  const SizedBox(width: 8),
                  Text(
                    isDeal ? '单' : label == '见人数' ? '人' : '次',
                    style: theme.textTheme.bodySmall?.copyWith(
                      color: theme.colorScheme.onSurfaceVariant,
                    ),
                  ),
                ],
              ),
              if (!locked && onRemove != null) ...[
                const SizedBox(height: 4),
                Align(
                  alignment: Alignment.centerRight,
                  child: TextButton(
                    onPressed: onRemove,
                    style: TextButton.styleFrom(
                      foregroundColor: Colors.red.shade400,
                    ),
                    child: const Text('移出基础任务', style: TextStyle(fontSize: 12)),
                  ),
                ),
              ],
            ] else if (!locked && onToggleIncluded != null) ...[
              const SizedBox(height: 4),
              Align(
                alignment: Alignment.centerRight,
                child: TextButton(
                  onPressed: onToggleIncluded,
                  child: const Text('加入基础任务', style: TextStyle(fontSize: 12)),
                ),
              ),
            ],
          ],
        ),
      ),
    );
  }
}

File: app/src/main/java/com/salesquest/sales_quest/ui/settings/TaskConfigPage.kt
package com.salesquest.sales_quest.ui.settings

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.interaction.MutableInteractionSource
import androidx.compose.foundation.interaction.collectIsFocusedAsState
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.text.BasicTextField
import androidx.compose.foundation.text.KeyboardOptions
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.ArrowBack
import androidx.compose.material.icons.filled.Celebration
import androidx.compose.material.icons.filled.CheckCircle
import androidx.compose.material.icons.filled.Groups
import androidx.compose.material.icons.filled.Info
import androidx.compose.material.icons.filled.Recommend
import androidx.compose.material.icons.filled.Search
import androidx.compose.material3.Button
import androidx.compose.material3.Card
import androidx.compose.material3.CenterAlignedTopAppBar
import androidx.compose.material3.CircularProgressIndicator
import androidx.compose.material3.ExperimentalMaterial3Api
import androidx.compose.material3.HorizontalDivider
import androidx.compose.material3.Icon
import androidx.compose.material3.IconButton
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.OutlinedButton
import androidx.compose.material3.Scaffold
import androidx.compose.material3.SnackbarHost
import androidx.compose.material3.SnackbarHostState
import androidx.compose.material3.Text
import androidx.compose.material3.TextButton
import androidx.compose.material3.TopAppBarDefaults
import androidx.compose.runtime.Composable
import androidx.compose.runtime.LaunchedEffect
import androidx.compose.runtime.collectAsState
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.rememberCoroutineScope
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.SolidColor
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.input.KeyboardType
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.lifecycle.viewmodel.compose.viewModel
import com.salesquest.sales_quest.core.DefaultTaskConfig
import kotlinx.coroutines.launch

/** 基础任务设置页 - 自定义目标 + 是否纳入成交 */
@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun TaskConfigPage(
    onBack: () -> Unit = {},
    viewModel: TaskConfigViewModel = viewModel()
) {
    val config by viewModel.config.collectAsState()
    val locked by viewModel.locked.collectAsState()
    val loading by viewModel.loading.collectAsState()
    val snackbarHostState = remember { SnackbarHostState() }
    val scope = rememberCoroutineScope()

    var meetTarget by remember { mutableStateOf(DefaultTaskConfig.recommendedMeetTarget) }
    var queryTarget by remember { mutableStateOf(DefaultTaskConfig.recommendedQueryTarget) }
    var dealTarget by remember { mutableStateOf(DefaultTaskConfig.recommendedDealTarget) }
    var includeMeet by remember { mutableStateOf(DefaultTaskConfig.recommendedIncludeMeet) }
    var includeQuery by remember { mutableStateOf(DefaultTaskConfig.recommendedIncludeQuery) }
    var includeDeal by remember { mutableStateOf(DefaultTaskConfig.recommendedIncludeDeal) }
    var saving by remember { mutableStateOf(false) }

    LaunchedEffect(config) {
        if (config != null) {
            meetTarget = config!!.meetTarget
            queryTarget = config!!.queryTarget
            dealTarget = config!!.dealTarget
            includeMeet = config!!.includeMeet
            includeQuery = config!!.includeQuery
            includeDeal = config!!.includeDeal
        }
    }

    Scaffold(
        topBar = {
            CenterAlignedTopAppBar(
                title = { Text("基础任务设置") },
                navigationIcon = {
                    IconButton(onClick = onBack) {
                        Icon(Icons.AutoMirrored.Filled.ArrowBack, contentDescription = "返回")
                    }
                },
                colors = TopAppBarDefaults.centerAlignedTopAppBarColors()
            )
        },
        snackbarHost = { SnackbarHost(snackbarHostState) }
    ) { innerPadding ->
        if (loading) {
            Box(
                modifier = Modifier
                    .fillMaxSize()
                    .padding(innerPadding),
                contentAlignment = Alignment.Center
            ) {
                CircularProgressIndicator()
            }
        } else {
            Column(
                modifier = Modifier
                    .fillMaxSize()
                    .padding(innerPadding)
                    .verticalScroll(rememberScrollState())
                    .padding(horizontal = 16.dp, vertical = 8.dp)
            ) {
                // === 锁定提示 ===
                InfoBanner(
                    text = if (locked) "今日任务已锁定 (已产生数据), 不可修改"
                    else "设置每日最低目标, 完成全部目标即可达成今日作战",
                    locked = locked
                )
                Spacer(Modifier.height(12.dp))

                // === 推荐目标按钮 ===
                if (!locked) {
                    OutlinedButton(
                        onClick = {
                            meetTarget = DefaultTaskConfig.recommendedMeetTarget
                            queryTarget = DefaultTaskConfig.recommendedQueryTarget
                            dealTarget = DefaultTaskConfig.recommendedDealTarget
                            includeMeet = DefaultTaskConfig.recommendedIncludeMeet
                            includeQuery = DefaultTaskConfig.recommendedIncludeQuery
                            includeDeal = DefaultTaskConfig.recommendedIncludeDeal
                        },
                        modifier = Modifier.fillMaxWidth()
                    ) {
                        Icon(Icons.Filled.Recommend, contentDescription = null)
                        Spacer(Modifier.size(8.dp))
                        Text("使用推荐目标 (见人100 / 查询5 / 成交不参与)")
                    }
                    Spacer(Modifier.height(12.dp))
                }

                // === 每日最低目标 ===
                Text(
                    "每日最低目标",
                    style = MaterialTheme.typography.titleSmall,
                    fontWeight = FontWeight.Bold,
                    color = MaterialTheme.colorScheme.primary,
                    modifier = Modifier.padding(vertical = 8.dp)
                )

                MetricConfigCard(
                    label = "见人数",
                    icon = Icons.Filled.Groups,
                    color = Color(0xFF2196F3),
                    included = includeMeet,
                    target = meetTarget,
                    locked = locked,
                    unit = "人",
                    onTargetChanged = { meetTarget = it },
                    onToggleInclude = { includeMeet = true },
                    onRemove = { includeMeet = false }
                )
                Spacer(Modifier.height(8.dp))

                MetricConfigCard(
                    label = "查询数",
                    icon = Icons.Filled.Search,
                    color = Color(0xFF9C27B0),
                    included = includeQuery,
                    target = queryTarget,
                    locked = locked,
                    unit = "次",
                    onTargetChanged = { queryTarget = it },
                    onToggleInclude = { includeQuery = true },
                    onRemove = { includeQuery = false }
                )
                Spacer(Modifier.height(8.dp))

                MetricConfigCard(
                    label = "成交数",
                    icon = Icons.Filled.Celebration,
                    color = Color(0xFFF44336),
                    included = includeDeal,
                    target = dealTarget,
                    locked = locked,
                    unit = "单",
                    isDeal = true,
                    onTargetChanged = { dealTarget = it },
                    onToggleInclude = { includeDeal = true },
                    onRemove = { includeDeal = false }
                )

                Spacer(Modifier.height(16.dp))

                // === 完成说明 ===
                CompletionHint()
                Spacer(Modifier.height(24.dp))

                // === 保存按钮 ===
                if (!locked) {
                    Button(
                        onClick = {
                            if (saving) return@Button
                            if (!includeMeet && !includeQuery && !includeDeal) {
                                scope.launch { snackbarHostState.showSnackbar("请至少选择一个指标") }
                                return@Button
                            }
                            saving = true
                            scope.launch {
                                try {
                                    viewModel.save(
                                        meetTarget = meetTarget,
                                        queryTarget = queryTarget,
                                        dealTarget = dealTarget,
                                        includeMeet = includeMeet,
                                        includeQuery = includeQuery,
                                        includeDeal = includeDeal
                                    )
                                    snackbarHostState.showSnackbar("已保存")
                                    onBack()
                                } catch (e: Exception) {
                                    snackbarHostState.showSnackbar("保存失败: ${e.message}")
                                } finally {
                                    saving = false
                                }
                            }
                        },
                        enabled = !saving,
                        modifier = Modifier
                            .fillMaxWidth()
                            .height(48.dp)
                    ) {
                        if (saving) {
                            CircularProgressIndicator(strokeWidth = 2.dp, modifier = Modifier.size(20.dp))
                        } else {
                            Text("保存")
                        }
                    }
                }
                Spacer(Modifier.height(32.dp))
            }
        }
    }
}

@Composable
fun InfoBanner(text: String, locked: Boolean) {
    val bg = if (locked) MaterialTheme.colorScheme.errorContainer
    else MaterialTheme.colorScheme.primaryContainer.copy(alpha = 0.3f)
    val fg = if (locked) MaterialTheme.colorScheme.onErrorContainer
    else MaterialTheme.colorScheme.primary
    Row(
        modifier = Modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(12.dp))
            .background(bg)
            .padding(12.dp),
        verticalAlignment = Alignment.CenterVertically
    ) {
        Icon(Icons.Filled.Info, contentDescription = null, tint = fg, modifier = Modifier.size(20.dp))
        Spacer(Modifier.width(8.dp))
        Text(text, style = MaterialTheme.typography.bodySmall, color = fg)
    }
}

@Composable
fun MetricConfigCard(
    label: String,
    icon: ImageVector,
    color: Color,
    included: Boolean,
    target: Int,
    locked: Boolean,
    unit: String,
    isDeal: Boolean = false,
    onTargetChanged: (Int) -> Unit,
    onToggleInclude: () -> Unit,
    onRemove: () -> Unit
) {
    Card(modifier = Modifier.fillMaxWidth()) {
        Column(modifier = Modifier.padding(12.dp)) {
            Row(verticalAlignment = Alignment.CenterVertically) {
                Icon(icon, contentDescription = null, tint = color, modifier = Modifier.size(22.dp))
                Spacer(Modifier.width(8.dp))
                Text(label, style = MaterialTheme.typography.bodyLarge, fontWeight = FontWeight.SemiBold)
                if (isDeal) {
                    Spacer(Modifier.width(6.dp))
                    Box(
                        modifier = Modifier
                            .clip(RoundedCornerShape(4.dp))
                            .background(Color(0xFFFF9800).copy(alpha = 0.15f))
                            .padding(horizontal = 6.dp, vertical = 2.dp)
                    ) {
                        Text("默认不参与", style = MaterialTheme.typography.labelSmall, color = Color(0xFFE65100))
                    }
                }
                Spacer(Modifier.weight(1f))
                Text(
                    if (included) "参与" else "不参与",
                    style = MaterialTheme.typography.labelSmall,
                    color = if (included) Color(0xFF4CAF50) else MaterialTheme.colorScheme.onSurfaceVariant,
                    fontWeight = FontWeight.Bold
                )
            }

            if (included) {
                Spacer(Modifier.height(12.dp))
                HorizontalDivider()
                Spacer(Modifier.height(12.dp))
                Row(verticalAlignment = Alignment.CenterVertically) {
                    Text("目标", style = MaterialTheme.typography.bodyMedium)
                    Spacer(Modifier.width(12.dp))
                    EditableTargetBox(
                        target = target,
                        color = color,
                        locked = locked,
                        onTargetChanged = onTargetChanged
                    )
                    Spacer(Modifier.width(8.dp))
                    Text(unit, style = MaterialTheme.typography.bodySmall, color = MaterialTheme.colorScheme.onSurfaceVariant)
                }
                if (!locked) {
                    Spacer(Modifier.height(4.dp))
                    TextButton(
                        onClick = onRemove,
                        modifier = Modifier.align(Alignment.End)
                    ) {
                        Text("移出基础任务", style = MaterialTheme.typography.labelSmall, color = Color(0xFFF44336))
                    }
                }
            } else if (!locked) {
                Spacer(Modifier.height(4.dp))
                TextButton(
                    onClick = onToggleInclude,
                    modifier = Modifier.align(Alignment.End)
                ) {
                    Text("加入基础任务", style = MaterialTheme.typography.labelSmall, color = MaterialTheme.colorScheme.primary)
                }
            }
        }
    }
}

/**
 * 可编辑的目标值控件 — 视觉上保留原数字显示 Box 风格,
 * 点击后可直接输入数字, 不使用标准表单输入框样式.
 */
@Composable
private fun EditableTargetBox(
    target: Int,
    color: Color,
    locked: Boolean,
    onTargetChanged: (Int) -> Unit
) {
    val interactionSource = remember { MutableInteractionSource() }
    val isFocused by interactionSource.collectIsFocusedAsState()
    var text by remember { mutableStateOf(target.toString()) }

    // 外部 target 变化时(配置加载 / 推荐目标), 同步显示文本
    LaunchedEffect(target) {
        if (!isFocused) {
            text = target.toString()
        }
    }

    // 失焦时校验并规范化输入
    LaunchedEffect(isFocused) {
        if (!isFocused) {
            val num = text.toIntOrNull()
            val validNum = num?.coerceIn(0, 9999) ?: 0
            text = validNum.toString()
            onTargetChanged(validNum)
        }
    }

    Box(
        modifier = Modifier
            .width(60.dp)
            .clip(RoundedCornerShape(8.dp))
            .background(color.copy(alpha = 0.1f))
            .then(
                if (isFocused && !locked) {
                    Modifier.border(1.dp, color, RoundedCornerShape(8.dp))
                } else {
                    Modifier
                }
            )
            .padding(vertical = 4.dp),
        contentAlignment = Alignment.Center
    ) {
        BasicTextField(
            value = text,
            onValueChange = { input ->
                val filtered = input.filter { it.isDigit() }
                if (filtered.length <= 4) {
                    text = filtered
                    val num = filtered.toIntOrNull()
                    if (num != null && num in 0..9999) {
                        onTargetChanged(num)
                    }
                }
            },
            enabled = !locked,
            singleLine = true,
            textStyle = MaterialTheme.typography.titleMedium.copy(
                fontWeight = FontWeight.Bold,
                color = color,
                textAlign = TextAlign.Center
            ),
            keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Number),
            cursorBrush = SolidColor(color),
            interactionSource = interactionSource,
            modifier = Modifier.fillMaxWidth(),
            decorationBox = { innerTextField ->
                Box(
                    modifier = Modifier.fillMaxWidth(),
                    contentAlignment = Alignment.Center
                ) {
                    innerTextField()
                }
            }
        )
    }
}

@Composable
fun CompletionHint() {
    Row(
        modifier = Modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(12.dp))
            .background(Color(0xFF4CAF50).copy(alpha = 0.08f))
            .padding(12.dp),
        verticalAlignment = Alignment.CenterVertically
    ) {
        Icon(Icons.Filled.CheckCircle, contentDescription = null, tint = Color(0xFF4CAF50), modifier = Modifier.size(20.dp))
        Spacer(Modifier.width(8.dp))
        Text(
            "完成以上全部目标即可完成今日作战\n连续完成每日基础任务 → 连续作战 +1",
            style = MaterialTheme.typography.bodySmall,
            color = Color(0xFF2E7D32)
        )
    }
}
